pkg/receive/config.go

```diff
@@ -21,6 +20,10 @@ import (
 	"gopkg.in/fsnotify.v1"
 )
 
+type EmptyHashringError string
+
+func (e EmptyHashringError) Error() string { return "hashring is empty" }
```

Comment:
This should not be a type but rather a new variable eg
```go
var EmptyHashringConfig = errors.New(...)
```

Reply:
I did this on purpose/ to use `errors.As` in the tests. I needed a type because it was more convenient to use `errors.As` and error types from `encoding/json` package. See `config_test.go`.

I didn't want to just test against the existence of an error, I wanted to be more specific.

Also even though go standard library uses package-level variables as sentinel errors, I'm not a fan. I think using constant errors or types is better for these type of errors.
 
All those being said, I also find it a bit ugly, however, I couldn't find a more elegant way, I'm open to suggestion.

Reply:
This is actually quite clean and easy to achieve thanks to the new errors package using errors.Is. Here is an example I put together: https://play.golang.org/p/3itSVfLjtNX

Reply:
Yes, for the known error from this package, this is a more elegant solution and easy to test with

However, the problem arises when you want to test against wrapped errors, which we use in `config_test.go`.  

```go
config, err := cw.parseConfig(cfgContent)
if err != nil {
	return nil, 0, errors.Wrap(err, "failed to load configuration file")
}
// ...

// config_test.go
// ...
{
	name: "unparsable config",
	cfg:  struct{}{},
	err:  json.UnsupportedTypeError{},
},

// ...
if err := cw.ValidateConfig(); err != nil && !errors.As(err, &tc.err) {
	t.Errorf("case %q: got unexpected error: %v", tc.name, err)
	continue
}
```
To test against wrapped error type I've used `errors.As`, hence the  empty config error's defined as a type.

Reply:
Note in the example I have you it IS a wrapped error. We wrap the FooError with `fmt.Errorf("bar: %w", err)`. errors.Is let's you check if the error that was wrapped is a specific error

Reply:
Yes, I'm well aware of that. I couldn't explain my problem clearly enough. I don't have the error value from the underlying `json` package, I only have the type that's why I couldn't use `errors.Is` for that specific case. 

I'll find another way to solve this, it's solely for testing anyway. 
Thanks for the feedback.